libs/SalesforceSDK/src/com/salesforce/androidsdk/accounts/UserAccount.java

```diff
@@ -452,7 +452,7 @@ public String getUserLevelStoragePath() {
 	 */
 	public String getCommunityLevelStoragePath() {
 		String leafDir = INTERNAL_COMMUNITY_PATH;
-		if (!TextUtils.isEmpty(communityId) && !communityId.equals(INTERNAL_COMMUNITY_ID)) {
+		if (!TextUtils.isEmpty(communityId) && !communityId.startsWith(INTERNAL_COMMUNITY_ID)) {
```

Comment:
Actually our app started normalizing 15 char community IDs to have AAA as the last 3 characters. That's how I found this problem, so not too unlikely 😉 

I thought this would be a safer approach since (I think) we try to make 15 or 18 char IDs interoperable everywhere. I believe through any server API, passing in 15 vs 18 char is required to work the same.